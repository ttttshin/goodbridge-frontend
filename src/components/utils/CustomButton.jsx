import React from "react";

// eslint-disable-next-line
export default function CustomButton({ text, action }) {
  return (
    <button
      onClick={action}
      className="bg-DarkGreen hover:bg-GoodBridge_Orange text-white font-style: italic py-2 px-4 rounded"
    >
      {text}
    </button>
  );
}
