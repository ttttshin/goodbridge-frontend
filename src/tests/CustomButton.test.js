import React from "react";
import { render } from "@testing-library/react";
import CustomButton from "../components/utils/CustomButton";

describe("Button component", () => {
  test("renders a button with given label", () => {
    const { asFragment } = render(<CustomButton label="Click me" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
