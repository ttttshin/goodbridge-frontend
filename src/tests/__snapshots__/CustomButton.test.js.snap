// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a button with given label 1`] = `
<DocumentFragment>
  <button
    class="bg-DarkGreen hover:bg-GoodBridge_Orange text-white font-style: italic py-2 px-4 rounded"
  />
</DocumentFragment>
`;
